### PR TITLE
apps: migrate to SPDX identifier

### PR DESCRIPTION
**Pull request by jerpelea**
## Summary
 Most tools used for compliance and SBOM generation use SPDX identifiers This change brings us a step closer to an easy SBOM generation.

## Impact
SBOM

## Testing
CI

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



No, this PR description does not meet the NuttX requirements.  While it attempts to follow the structure, it lacks crucial details.  Here's why:

* **Summary:**  While it states the *why*, it's vague. What tools? What kind of SPDX identifiers?  It completely omits *what* is changing, *how* the change works, and any related issue references.  Example: "This change adds SPDX license identifiers to the source files within the core OS directory (e.g., `sched/`, `drivers/`). This will allow tools like [tool name] to generate a compliant SBOM.  This addresses issue #[issue number]."
* **Impact:** Simply stating "SBOM" isn't helpful.  You need to address *all* impact points with "YES/NO" and descriptions where necessary.  For example,  "Impact on documentation: YES - The README will be updated to explain how to generate an SBOM."  Even if the answer is "NO", state it explicitly.
* **Testing:**  "CI" isn't sufficient. List the specific build hosts and targets tested.  Provide *actual* log snippets demonstrating the change's effect, not just "your testing logs here".  Show the absence of SPDX identifiers before and their presence after the change.  Example:
    * Build Host(s): Linux (Ubuntu 22.04), x86_64, GCC 11.2
    * Target(s): qemu-sim:default

    Testing logs before change (excerpt from `sched/sched.c`):
    ```c
    /* No SPDX identifier present */
    ```

    Testing logs after change (excerpt from `sched/sched.c`):
    ```c
    // SPDX-License-Identifier: Apache-2.0
    ```

In short, the PR description needs to be significantly more detailed and explicit to meet the NuttX requirements.  It should provide specific information about the changes, their impact, and clear evidence of testing.

**Comment by jerpelea** [comment]
@xiaoxiang781216 please ignore can nxerrors

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by cederom — APPROVED** [review]
Thank you @jerpelea :-)

**Comment by cederom** [comment]
build errors come from previously merged pr? o_O

File: builtin/builtin_list.c
Comment (cederom) [inline builtin/builtin_list.c:1]: code change, potential build error cause.

**Review by cederom — CHANGES_REQUESTED** [review]
* Some code changes slipped in causing build errors that needs to be fixed?
* Maybe also good moment to fix reported code formatting issues?
* Or just update SPDX headers without touching the code?

File: builtin/builtin_list.c [hunk 1]
```diff
@@ -41,11 +46,8 @@
  * Public Data
  ****************************************************************************/
 
-#include "builtin_proto.h"
-
 const struct builtin_s g_builtins[] =
 {
-#  include "builtin_list.h"
```
Comment (xiaoxiang781216) [inline builtin/builtin_list.c:48]: should not change

**Comment by jerpelea** [comment]
@xiaoxiang781216 ready to go

**Review by cederom — APPROVED** [review]
Thank you @jerpelea :-)

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**